core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java

```diff
@@ -1259,6 +1259,259 @@ protected final MaterializedViewFixture sql(String materialize,
+        + "order by \"deptno\"\n"
+        + "offset 3 rows";
+    sql(mv, query).noMat();
+  }
+
+  /**
+   * It need be matched, info of offset and fetch.
+   * eg: mv[2, null], query[3, 3].
+   */
+  @Test void testSortToSort6() {
+    final String mv = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"\n"
+        + "offset 2 rows";
+    final String query = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"\n"
+        + "offset 3 rows\n"
+        + "fetch next 3 rows only";
+    sql(mv, query)
+        .checkingThatResultContains(""
+            + "LogicalSort(offset=[1], fetch=[3])\n"
+            + "  EnumerableTableScan(table=[[hr, MV0]])")
+        .ok();
+  }
+
+  /**
+   * It need be matched, info of offset and fetch.
+   * eg: mv[2, 10], query[3, 3]
+   */
+  @Test void testSortToSort7() {
+    final String mv = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"\n"
+        + "offset 2 rows\n"
+        + "fetch next 10 rows only";
+    final String query = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"\n"
+        + "offset 3 rows\n"
+        + "fetch next 3 rows only";
+    sql(mv, query)
+        .checkingThatResultContains(""
+            + "LogicalSort(offset=[1], fetch=[3])\n"
+            + "  EnumerableTableScan(table=[[hr, MV0]])")
+        .ok();
+  }
+
+  /**
+   * It needn't be matched, info of offset and fetch.
+   * eg: mv[5, 10], query[3, 3]
+   */
+  @Test void testSortToSort8() {
+    final String mv = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"\n"
+        + "offset 5 rows\n"
+        + "fetch next 10 rows only";
+    final String query = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"\n"
+        + "offset 3 rows\n"
+        + "fetch next 3 rows only";
+    sql(mv, query).noMat();
+  }
+
+  /**
+   * It need be matched, query has direction's info, which cloud rebuild by mv.
+   */
+  @Test void testSortToSort9() {
+    final String mv = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"";
+    final String query = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\" desc";
+    sql(mv, query)
+        .checkingThatResultContains(""
+            + "LogicalSort(sort0=[$1], dir0=[DESC])\n"
+            + "  EnumerableTableScan(table=[[hr, MV0]])")
+        .ok();
+  }
+
+  /**
+   * It need be matched, query has direction's info, which cloud rebuild by mv.
+   * And query also has the info of offset and fetch.
+   */
+  @Test void testSortToSort10() {
+    final String mv = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
```

Comment:
how about sort multiple columns ？At the same time, the order of columns is different.

Reply:
Ascending or descending order？

Reply:
Got it, I add multiple-columns-sort test case.